src/services/storage-service.js

```diff
@@ -7,25 +6,28 @@ import UserStore from '../stores/user-store.js';
 import RoomStore from '../stores/room-store.js';
 import LockStore from '../stores/lock-store.js';
 import LessonStore from '../stores/lesson-store.js';
+import ServerConfig from '../bootstrap/server-config.js';
 import StoragePlanStore from '../stores/storage-plan-store.js';
 import TransactionRunner from '../stores/transaction-runner.js';
 import { componseUniqueFileName } from '../utils/path-utils.js';
 import RoomInvitationStore from '../stores/room-invitation-store.js';
 import permissions, { hasUserPermission } from '../domain/permissions.js';
-import { ROOM_ACCESS_LEVEL, ROOM_LESSONS_MODE, STORAGE_LOCATION_TYPE } from '../domain/constants.js';
-import { getPrivateStoragePathForRoomId, getPrefixFromStoragePath, getStoragePathType, STORAGE_PATH_TYPE } from '../ui/path-helper.js';
+import { CDN_OBJECT_TYPE, ROOM_ACCESS_LEVEL, ROOM_LESSONS_MODE, STORAGE_LOCATION_TYPE } from '../domain/constants.js';
+import { getPrivateStoragePathForRoomId, getPrefixFromStoragePath, getStoragePathType, STORAGE_PATH_TYPE, getPathSegments } from '../ui/path-helper.js';
 
 const { BadRequest } = httpErrors;
 
 export default class StorageService {
-  static get inject() { return [Cdn, RoomStore, RoomInvitationStore, LessonStore, StoragePlanStore, UserStore, LockStore, TransactionRunner]; }
+  static get inject() { return [ServerConfig, Cdn, RoomStore, RoomInvitationStore, LessonStore, StoragePlanStore, UserStore, LockStore, TransactionRunner]; }
 
-  constructor(cdn, roomStore, roomInvitationStore, lessonStore, storagePlanStore, userStore, lockStore, transactionRunner) {
+  // eslint-disable-next-line max-params
```

Comment:
I think the mapping should be done here and not in the controller, but then we need the server config as an additional argument) :-(

Reply:
I think that's fine, unless we make a dedicated method in the `client-data-mapping-service`. That way would be more aligned with all other client data we map.